game/songfilter.cc

```diff
@@ -0,0 +1,113 @@
+#include "songfilter.hh"
+
+#include "song.hh"
+#include "util.hh"
+
+#include <iostream>
+
+namespace {
+	bool matches(std::string const& word, std::set<std::string> const& terms) {
+		for(auto const& term : terms) {
+			if(containsNoCase(word, term))
+				return true;
+		}
+
+		return false;
+	}
+}
+
+GenreFilter::GenreFilter(std::string const& genre, std::set<std::string> const& genres)
+: m_genre(genre), m_genres(genres) {
+}
+
+bool GenreFilter::filter(Song const& song) const {
+	if(matches(song.genre, m_genres))
+		return containsNoCase(song.genre, m_genre);
+
+	return m_genre == "Other";
+}
+
+
+LanguageFilter::LanguageFilter(std::string const& language)
+: m_language(language) {
+	//std::cout << "set language " << language << std::endl;
```

Comment:
Commented prints `std::cout` can be removed or replaced with debug/trace log messages